src/commands/deploy/deploy.ts

```diff
@@ -118,6 +119,12 @@ const getDeployFolder = async ({
   }
 
   if (!deployFolder) {
+    if (!stdin.isTTY || !stdout.isTTY) {
```

Comment:
nit: Might be nice to abstract this to a `isInteractive` method that we use in the different places we're checking this to ensure consistent behaviour. I could also see us introducing an explicit flag to control this behaviour (similar to npm's `--yes`) and that would make it easier to integrate. Not a blocker, though.